packages/js/src/server/integrations/uncaught_exception.ts

```diff
@@ -14,28 +15,58 @@ function removeAwsLambdaListener() {
   removeAwsDefaultUncaughtExceptionListener()
 }
 
+/**
+ * If there are no other uncaughtException listeners,
+ * we want to report the exception to Honeybadger and
+ * mimic the default behavior of NodeJs,
+ * which is to exit the process with code 1
+ */
+function hasOtherUncaughtExceptionListeners() {
+  return process.listeners('uncaughtException').length > 1
+}
+
 export default function (): Types.Plugin {
   return {
     load: (client: typeof Client) => {
-      if (!client.config.enableUncaught) { return }
+      if (!client.config.enableUncaught) {
+        return
+      }
 
       removeAwsLambdaListener()
 
+
       process.on('uncaughtException', function honeybadgerUncaughtExceptionListener(uncaughtError) {
-        // Prevent recursive errors
-        if (count > 1) { fatallyLogAndExit(uncaughtError) }
-
-        if (client.config.enableUncaught) {
-          client.notify(uncaughtError, {
-            afterNotify: (_err, _notice) => {
-              count += 1
-              client.config.afterUncaught(uncaughtError)
-            }
-          })
-        } else {
-          count += 1
+        if (!client.config.enableUncaught) {
           client.config.afterUncaught(uncaughtError)
+          if (!hasOtherUncaughtExceptionListeners()) {
+            fatallyLogAndExit(uncaughtError)
+          }
+          return
+        }
+
+        // report only the first error - prevent reporting recursive errors
+        if (handlerAlreadyCalled) {
+          if (!hasOtherUncaughtExceptionListeners()) {
+            fatallyLogAndExit(uncaughtError)
+          }
+          return
+        }
+
+        if (isReporting) {
+          return
         }
```

Comment:
@shalvah

Reply:
Nice👍

Reply:
(Do you know if there's a reason why the rejection handler doesn't do this?)